src/main/scala/org/labrad/registry/Registry.scala

```diff
@@ -215,38 +228,57 @@ extends LocalServer with Logging {
       data.convertTo(pattern)
     }
 
-    @Setting(id=30,
-             name="set",
-             doc="Set the content of the given key in the current directory to the given data")
+    @Setting(id = 30,
+             name = "set",
+             doc = """Set the given key in the current directory to the given value.""")
     def setValue(key: String, value: Data): Unit = {
       require(key.nonEmpty, "Cannot create a key with empty name")
       store.setValue(curDir, key, value)
     }
 
-    @Setting(id=40,
-             name="del",
-             doc="Delete the given key from the current directory")
+    @Setting(id = 40,
+             name = "del",
+             doc = """Delete the given key from the current directory.""")
     def delete(key: String): Unit = {
       store.delete(curDir, key)
     }
 
-    @Setting(id=50,
-             name="Notify on Change",
-             doc="Requests notifications if the contents of the current directory change")
+    @Setting(id = 50,
+             name = "Notify on Change",
+             doc = """Requests notifications if the contents of the current directory change.
+                 |
+                 |Called with message id and a flag indicating whether notifications should be
+                 |enabled or disabled. Notification messages will be sent to the given id in the
+                 |context used when calling this setting. The message will be of the form
+                 |(name, isDir, addOrChange), where name is the name of an item in the current
+                 |directory, isDir is a boolean indicating whether the item is a directory (true) or
+                 |key (false), and addOrChange is a boolean indicating whether the item was added or
+                 |changed (true), or was deleted (false).""")
     def notifyOnChange(r: RequestContext, id: Long, enable: Boolean): Unit = {
       changeListener = if (enable) Some((r.source, id)) else None
     }
 
-    @Setting(id=55,
-             name="Stream Changes",
-             doc="Requests notifications of all changes made in any directory")
+    @Setting(id = 55,
+             name = "Stream Changes",
+             doc = """Requests notifications of all changes made in any directory.
+                 |
```

Comment:
Yeah, all changes underneath the current path would be more useful, and could easily replicate the current behavior just by sitting in the root directory.